### PR TITLE
ISLRU activation function

### PR DESCRIPTION
**Pull request by abh2k**

### CODE REVIEW

**Comment by abh2k** [comment]
I accidently forced push a wrong commit in  my last PR, i am not sure what happened.    Luckily  the code on my  forked branch was not affected, so I created a new PR.

File: src/mlpack/methods/ann/layer/isrlu.hpp [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+ * @author Abhinav Anand
+ *
+ * Definition of the ISRLU activation function as described by Jonathan T. Barron.
+ *
+ * For more information, read the following paper.
+ *
+ * @code
+ * @article{
+ *   author  = {Carlile, Brad and Delamarter, Guy and Kinney, Paul and Marti,
+ *              Akiko and Whitney, Brian},
```
Comment (birm) [inline src/mlpack/methods/ann/layer/isrlu.hpp:12]: Unless there's a reason to have the names in those pairings that I'm unaware of, just list them.
```suggestion
 *   author  = {Carlile, Brad, Delamarter, Guy, Kinney, Paul, Marti, Akiko,
 *              Whitney, Brian},
```

**Comment by birm** [comment]
This appears to be a follow up to #2830

File: src/mlpack/methods/ann/layer/isrlu.hpp (hunk 1 above)
Comment (birm) [reply src/mlpack/methods/ann/layer/isrlu.hpp:12]: Whoops, nevermind.

Comment (birm) [inline src/mlpack/methods/ann/layer/isrlu.hpp:12]: It _looks_ like mlpack prefers to have lists of authors in First Last, First Last ... format. I think that may help with readability.

Comment (abh2k) [reply src/mlpack/methods/ann/layer/isrlu.hpp:12]: So shall i remove `and` from the names.

Comment (birm) [reply src/mlpack/methods/ann/layer/isrlu.hpp:12]: A suggestion, if you find it helpful/to your liking
```suggestion
 *   author  = {Brad Carlile, Guy Delamarter, Paul Kinney, Akiko Marti,
 *              and Brian Whitney},
```

File: src/mlpack/methods/ann/layer/isrlu.hpp [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+ * @author Abhinav Anand
+ *
+ * Definition of the ISRLU activation function as described by Jonathan T. Barron.
+ *
+ * For more information, read the following paper.
+ *
+ * @code
+ * @article{
+ *   author  = {Carlile, Brad, Delamarter, Guy, Kinney, Paul, Marti,
+ *              Akiko, Whitney, Brian},
```
Comment (birm) [inline src/mlpack/methods/ann/layer/isrlu.hpp:12]: sorry for the confusion. This was my original suggestion when I thought that these were all last names, as opposed to first and last names.
I'd again recommend a format which makes it clear. Maybe you'd prefer just last names?

```suggestion
 *   author  = {Carlile, Delamarter, Kinney, Marti, Whitney},
```

Comment (rcurtin) [reply src/mlpack/methods/ann/layer/isrlu.hpp:12]: Another way to get first names correctly in BibTeX format would be something like `Carlile, Brad and Delamarter, Guy and Kinney, Paul and Marti, Akiko and Whitney, Brian}`. :+1:

Comment (birm) [reply src/mlpack/methods/ann/layer/isrlu.hpp:12]: ^ This whole mess is because I found that format confusing. It must just be me then

Comment (rcurtin) [reply src/mlpack/methods/ann/layer/isrlu.hpp:12]: Yeah, the BibTeX format is confusing but I guess I've gotten used to it.  I think you can also do `Brad Carlile and Guy Delamarter and Paul Kinney and ...`, but I am not 100% sure.  Maybe that's a little better? :smile:

Comment (birm) [reply src/mlpack/methods/ann/layer/isrlu.hpp:12]: I'll add both suggestions, in hope that that's convient for @abh2k 
```suggestion
 *   author  = {Brad Carlile and Guy Delamarter and Paul Kinney and Marti Akiko
 *              and Brian Whitney},
```

Comment (birm) [reply src/mlpack/methods/ann/layer/isrlu.hpp:12]:
```suggestion
 *   author  = {Carlile, Brad and Delamarter, Guy and Kinney, Paul and Marti,
 *              Akiko and Whitney, Brian},
```
(Sorry for being confused by this earlier!)

File: src/mlpack/tests/activation_functions_test.cpp [hunk 3]
```diff
@@ -991,6 +1039,22 @@ TEST_CASE("CELUFunctionTest", "[ActivationFunctionsTest]")
   CheckCELUDerivativeCorrect(desiredActivations, desiredDerivatives);
 }
 
+/**
+ * Basic test of the ISRLU activation function.
+ */
+TEST_CASE("ISRLUFunctionTest", "[ActivationFunctionsTest]")
+{
+  const arma::colvec desiredActivations("-0.89442719 3.2 4.5 \
```
Comment (birm) [inline src/mlpack/tests/activation_functions_test.cpp:1047]: Where did you get/calculate these test values?

Comment (abh2k) [reply src/mlpack/tests/activation_functions_test.cpp:1047]: https://arxiv.org/abs/1710.09967 This is  the paper.   The   input value is same for all activation test cases which can be found at the beginning of the file and the desired activation can  be calculate          using the formula in the paper.     I just  used                a high       precision calculator and  calculated the values.

**Review by birm — APPROVED** [review]
Looks good to me, thank you for the new layer type!

**Review by zoq — APPROVED** [review]
Looks good to me as well, thanks for keeping up with the comments.

**Pull request merged**